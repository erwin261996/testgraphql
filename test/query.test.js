const { gql } = require("apollo-server");
const createTestServer = require("./helper");
const models = require("../models")

const USER = gql `
    {
        allusers {
            firstName,
            lastName
            id
        }
    }
`

describe('queries', () => {
    test('allusers', async () => {
        const {query} = createTestServer({
            "allusers": jest.fn(models.User.findAll())            
        })
        // https://github.com/erwin261996/testgraphql.git

        // const res = await query({ query: USER, http: { headers: { authorization: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjIsImVtYWlsIjoidGVzdDJAZ21haWwuY29tIiwicm9sZSI6IkFETUlOIiwiaWF0IjoxNjQwMjk1MzY3LCJleHAiOjE2NDAzMDI1Njd9.5yXc6aDKnAP6W8glTuCqwSU0Uq_0A7roPRMMnIRpEMc` } },})
        const res = await query({ query: USER })
        expect(res).toMatchSnapshot()
    })
})