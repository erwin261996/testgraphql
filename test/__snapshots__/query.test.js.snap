// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries User 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: not authorized],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries allusers 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Cannot read properties of undefined (reading 'User')],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
